src/Builder/DatagridBuilder.php

```diff
@@ -163,7 +164,7 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = array())
             $defaultOptions['csrf_protection'] = false;
         }
 
-        $formBuilder = $this->formFactory->createNamedBuilder('filter', 'form', array(), $defaultOptions);
+        $formBuilder = $this->formFactory->createNamedBuilder('filter', FormType::class, array(), $defaultOptions);
```

Comment:
I'm afraid this has to be conditional, if you're supporting sf 2.3 . See https://github.com/sonata-project/SonataClassificationBundle/pull/315